libraries/libraries.mk

```diff
@@ -130,25 +130,45 @@ $(LIB_STRICT_OBJECTS): CXXFLAGS += -Wno-unused-variable
 $(LIB_STRICT_OBJECTS): CXXFLAGS += -Wno-unused-function
 $(LIB_STRICT_OBJECTS): CXXFLAGS += -Wno-unused-but-set-variable
 
+ifdef __NEWLIB
+LIB_OBJECTS += $(patsubst %S,%o,$(LIB_SSOURCES))
+$(BSG_PLATFORM_PATH)/libbsg_manycore_runtime.so.1.0: $(LIB_OBJECTS)
+	$(RV_AR) -rcs $@ $^
+else
 $(BSG_PLATFORM_PATH)/libbsg_manycore_runtime.so.1.0: LD = $(CXX)
 $(BSG_PLATFORM_PATH)/libbsg_manycore_runtime.so.1.0: $(LIB_OBJECTS)
 	$(LD) -shared -Wl,-soname,$(basename $(notdir $@)) -o $@ $^ $(LDFLAGS)
+endif
```

Comment:
I'm concerned that newlib specific code is showing up in a platform independent make fragment

Reply:
Yes, you are right about using `ar` for static libraries. For BlackParrot, we need to have static linking of libraries since we can't do dynamic linking. 

The newlib-specific code is showing up here because we want to compile the library with newlib as well. The platforms so far all relied on x86 and now that we have another ISA in the mix, something like this has to be done. Probably, this might not be the best way to do it, but I am afraid this sort of stuff will creep into the platform independent make file, however small it may be.

Reply:
This generates a static library with a .so suffix right?

Reply:
Yeah. I get that it is confusing but this was the only way I could think of that was minimally invasive to the platform independent make fragment.

Reply:
Could you define a .a target in the platform library fragment?

Reply:
I believe that is already there here - https://github.com/sripathi-muralitharan/bsg_replicant/blob/8ff070199315e0ac623e1789c31fb7743f4f652d/libraries/platforms/dromajo-vcs/library.mk#L134

Reply:
Then why not just link it there?

There seems to be part of this that I'm missing.

Reply:
If I have to do that, then I should include the platform independent make fragment in the platform-specific one making it recursive includes. I would have to redefine the same variables (LIB_OBJECTS, etc..) in the platform-specific version creating duplication and making it harder to maintain.

Reply:
So the issue is that you want to access some of the variables in the independent makefile, but you can't?

Reply:
Or as issue, at least

Reply:
I can't without redefining.

Reply:
I think I'm starting to understand the issues here.

Why __NEWLIB? I think I have an idea for you.

Reply:
Removed all of this. Added relevant code for the suggested changes.